### PR TITLE
Tweak online publication article specification (DENNER-2431)

### PR DESCRIPTION
**Pull request by curzio-della-santa**

### CODE REVIEW

**Comment by curzio-della-santa** [comment]
Should update example of online-publication (that contains the articles) too,.
I'll do it as soon as this PR is approved

File: src/swagger.yml [hunk 1]
```diff
@@ -4300,16 +4479,40 @@ definitions:
...
+        allOf:
+          - $ref: '#/definitions/Translations'
+          - description: >
+              Localized information on single unit price, currently available for a 'bottle' (wines) or a 'package' (cigarettes).
+              Note: might be deprecated in future in favor of a 'unit_price' object containing the placeholders and message.
+#      panel: Does not exists anymore
+#        $ref: '#/definitions/Translations'
+      stopper:
+        $ref: '#/definitions/Stopper'
+      availability_limitations:
```
Comment (curzio-della-santa) [inline src/swagger.yml]: swap with dots (I'm changing the order in this exact moment)

File: src/swagger.yml [hunk 2]
```diff
@@ -4300,16 +4479,40 @@ definitions:
...
+        allOf:
+          - $ref: '#/definitions/Translations'
+          - description: >
+              Localized instead price string for display (includes 'instead' translation
+              and price might contain a zero lead dash or a zero decimals dash)
+      price_info:
+        allOf:
+          - $ref: '#/definitions/Translations'
+          - description: Localized information on price (currently can be the localized texts for 'compared')
+      unit_price_info:
```
Comment (curzio-della-santa) [inline src/swagger.yml]: Substitute with 'unit_price' as boolean

File: examples/online-publication-article.json [hunk 3]
```diff
@@ -133,96 +386,78 @@
...
-            "locale": "de",
-            "value": "9.- sparen"
-        },
-        {
-            "locale": "fr",
-            "value": "9.- éparmié"
         },
         {
-            "locale": "it",
-            "value": "9.- risparmiati"
```
Comment (ivan-wolf) [inline examples/online-publication-article.json]: This is a breaking change!

File: examples/online-publication-article.json [hunk 4]
```diff
@@ -133,96 +386,78 @@
...
-            {
-                "locale": "fr",
-                "value": "..."
-            },
-            {
-                "locale": "it",
-                "value": "..."
-            }
-        ]
-    },
```
Comment (ivan-wolf) [inline examples/online-publication-article.json]: This is a breaking change!

File: examples/online-publication-article.json [hunk 5]
```diff
@@ -19,94 +20,338 @@
...
+        },
+        {
+            "locale": "it",
+            "value": "I prezzi promozionali delle singole bottiglie sono validi solo nella rispettiva settimana promozionale"
+        }
+    ],
+    "packshot": {
+        "id": "c3391581-1159-4611-a5ca-d15cc27e79dc",
+        "quantity": 6,
+        "url": "https://denner-assets.s3.amazonaws.com/assets/c3391581-1159-4611-a5ca-d15cc27e79dc/web/051273_6.png",
```
Comment (ivan-wolf) [inline examples/online-publication-article.json:231]: Make an example with a TIFF here?

File: examples/online-publication-article.json [hunk 6]
```diff
@@ -19,94 +20,338 @@
...
+        "image_url": "https://denner-assets.s3.amazonaws.com/assets/c3391581-1159-4611-a5ca-d15cc27e79dc/web/051273_6.png",
+        "preview_url": "https://denner-assets.s3.amazonaws.com/assets/c3391581-1159-4611-a5ca-d15cc27e79dc/web/051273_6.png?version=1",
```
Comment (ivan-wolf) [inline examples/online-publication-article.json:233]: Remove `image_url` and `preview_url`?

File: examples/online-publication-article.json [hunk 7]
```diff
@@ -19,94 +20,338 @@
...
+        "cdn_url": "https://denner.imgix.net/assets/c3391581-1159-4611-a5ca-d15cc27e79dc/web/051273_6.png",
+        "archived": false,
+        "view": {
+            "id": "front"
+        }
```
Comment (ivan-wolf) [inline examples/online-publication-article.json:238]: Aren't there additonal properties like `year` and `created_on`?

File: examples/online-publication-article.json [hunk 8]
```diff
@@ -491,40 +680,106 @@
...
+                        "final": true
+                    },
+                    {
+                        "quantity": 1,
+                        "sales": 11.4,
+                        "final": true
+                    }
+                ]
+            },
+            "discount": [],
```
Comment (ivan-wolf) [inline examples/online-publication-article.json:791]: Add an example?

File: examples/online-publication-article.json [hunk 9]
```diff
@@ -1,7 +1,8 @@
 {
-    "id": "dfbc2b0d-0f47-4ac2-86e5-3bd4f9936ef6",
+    "id": "938f5d58-e9b0-439f-b5da-5cf086b5e412",
     "article_id": "051273",
-    "name": "Mandorla Primitivo IGT Puglia",
+    "advertised_article_id": "3ef799aa-3202-471f-8cc1-79f0293923d0",
```
Comment (ivan-wolf) [inline examples/online-publication-article.json:4]: Really expose this?

File: examples/online-publication-article.json [hunk 10]
```diff
@@ -19,94 +20,338 @@
...
+    },
+    "purchasing_department": "W",
+    "online_group": {
+        "id": "a2664d3a-93b4-4277-9e6e-c95ad8b6c6ee",
+        "code": "2-9",
+        "sort_index": 1,
+        "disabled": false,
+        "allow_featuring": true,
+        "allow_promotion": true,
+        "allow_background": true,
```
Comment (ivan-wolf) [inline examples/online-publication-article.json:113]: Really expose `disabled` and these `allow_*` properties?

File: examples/online-publication-article.json [hunk 11]
```diff
@@ -400,36 +568,57 @@
...
+                        "locale": "fr",
+                        "value": "Online Specials"
+                    },
+                    {
+                        "locale": "it",
+                        "value": "Online Specials"
                     }
                 ]
             }
         ],
```
Comment (ivan-wolf) [inline examples/online-publication-article.json:657]: The new property `availability` was added in the meantime.

File: examples/online-publication-article.json [hunk 12]
```diff
@@ -400,36 +568,57 @@
         },
         "online_specials": [
             {
-                "id": "rarity",
+                "id": "bordeaux-stationary",
+                "name": "Bordeaux Grand Cru (Online & stationär)",
```
Comment (ivan-wolf) [inline examples/online-publication-article.json:621]: The name and titles changed in the meantime.

File: examples/online-publication-article.json [hunk 13]
```diff
@@ -491,40 +680,106 @@
...
+                    "value": "Test <b>FR</b>"
                 },
                 {
                     "locale": "it",
-                    "value": "Attualmente non disponibile"
+                    "value": "Test <b>IT</b>"
                 }
             ]
-        },
-        "shop_url": null
```
Comment (ivan-wolf) [inline examples/online-publication-article.json:514]: This is a breaking change!

**Review by ivan-wolf — CHANGES_REQUESTED** [review]
@curzio-della-santa, I guess we should discuss this...

**Comment by curzio-della-santa** [comment]
I've merged master into this .. to not get too much in trouble when I'll finish this  PR

**Pull request closed without merging**